### PR TITLE
ENH: Questionnaire Plugin

### LINKED ISSUE #30: Include Questionaire Loads Through Happi
See title

### PR DESCRIPTION
**Pull request by teddyrendahl**
## Description

Use the `QSBackend` to load information from the PCDS Questionnaire. Depends on the `experiment` section of the YAML file, specifically `proposal` and `run`

## Motivation and Context

Closes #30
## How Has This Been Tested?

Added a slightly convoluted test to load a custom device that allows us to check the internals of the Backend.

## Where Has This Been Documented?

Module level docstring added to Sphinx

### CODE REVIEW

File: hutch_python/plugins/questionnaire.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        proposal = self.conf['experiment'].get('proposal')
+        if run and proposal:
+            logger.debug("Loading Questionnaire information for %s in Run %s",
+                         proposal, run)
+            qs_client = happi.Client(database=QSBackend(run, proposal))
+        else:
+            raise ValueError("Inadequate information to load Questionnaire. "
+                             "Must specify proposal and run.")
+        dev_namespace = load_devices(*qs_client.all_devices, pprint=True)
+        return dev_namespace.__dict__
```
Comment (ZLLentz) [inline hutch_python/plugins/questionnaire.py]: Implementing the bulk of this in `happi` really paid off, this is very clean for an interface layer.

File: hutch_python/tests/test_plugins/test_questionnaire.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        else:
+            return devices[0]
+
+
+def test_questionnaire_plugin():
+    logger.debug("test_questionnaire_plugin")
+    conf = {'experiment': {'run': '15', 'proposal': 'LR12'},
+            'questionnaire': True}
+    plugin = Plugin(conf)
+    hutch_python.plugins.questionnaire.QSBackend = QSBackend
```
Comment (ZLLentz) [inline hutch_python/tests/test_plugins/test_questionnaire.py]: sneaky 👍

File: hutch_python/tests/conftest.py [hunk 3]
```diff
@@ -30,3 +31,6 @@ def restore_logging():
     prev_handlers = copy(logging.root.handlers)
     yield
     logging.root.handlers = prev_handlers
+
+
+Experiment = namedtuple('Experiment', ('run', 'proposal'))
```
Comment (ZLLentz) [inline hutch_python/tests/conftest.py]: Where is this `namedtuple` used?

Comment (teddyrendahl) [reply hutch_python/tests/conftest.py]: My fake object that gets created by `load_devices`. I wanted to a full end to end test, so the fake Questionnaire info looks likes
```python
device_class = 'hutch_python.tests.conftest'
args = ['{{run}}', '{{proposal}}']
```
This way when we run `load_devices`, `Experiment` gets imported and instantiated.

File: hutch_python/tests/test_plugins/test_questionnaire.py (hunk 2 above)
Comment (teddyrendahl) [reply hutch_python/tests/test_plugins/test_questionnaire.py]: With great Python power comes great responsibility ... Crazy this works.

**Comment by teddyrendahl** [comment]
@ZLLentz The YAML loader will automatically convert "True" to a boolean from a string.

**Pull request merged**